tests/robot/_resources/libraries/jsonlib.py

```diff
@@ -131,13 +123,62 @@ def compare_jsons(
     for change in changes:
         if change in diff:
             change_counter += 1
-            logger.debug(
-                "{}. CHANGE ({}): {}".format(change_counter, change, diff[change])
-            )
+            logger.debug(f"{change_counter}. CHANGE ({change}): \n{diff[change]}\n\n")
 
     return diff.to_dict()
 
 
+def ignore_type_properties(obj, path):
+    """
+    This is a callback function. It is used inside of `compare_jsons_ignoring_properties()`.
+    It takes the object and its path and returns a Boolean. If True is returned,
+    the object is excluded from the results, otherwise it is included.
+    `obj` refers to the value part of a key:value pair.
+    `path` refers to the location of obj in the dict.
+    """
+    ignorable_types = [
+        # "ACTIVITY",
+        "ARCHETYPED",
```

Comment:
@testautomation the _type inside the class on the data values (DV) doesn't depend on the type, but on the parent. For instance, if the data value is inside ELEMENT.value, the _type is required because "value" can be any data value. This is key to correctly checking the _type should be there or not. So what I would to is a list like this with the parent attributes:

COMPOSITION.context
COMPOSITION.archetype_details // this is inherited from LOCATABLE, so this should be repeated for each LOCATABLE subclass (SECTION, OBSERVATION, EVALUATION, ... CLUSTER, ELEMENT)
COMPOSITION.archetype_details.archetype_id
COMPOSITION.archetype_details.template_id
COMPOSITION.archetype_details.archetype_id
SECTION.archetype_details
SECTION.archetype_details.archetype_id
SECTION.archetype_details.template_id
....

So, at those points, the _type is not required.




This list seems to have missing data values:

- DV_BOOLEAN
- DV_STATE
- DV_IDENTIFIER
- TERM_MAPPING
- DV_ORDINAL
- DV_SCALE
- REFERENCE_RANGE
- DV_PROPORTION
- DV_QUANTITY
- DV_DURATION
- DV_TIME
- DV_DATE
- DV_DATE_TIME
- DV_PARSABLE
- DV_MULTIMEDIA
- DV_URI
- DV_EHR_URI

Reply:
@ppazos I've added suggested DVs to the list but will need a call w/ you to clarify if I understood your comment correctly.